Runtime/Plugins/Raycast/RaycastableContainer.cs

```diff
@@ -0,0 +1,118 @@
+using Unity.UIWidgets.foundation;
+using Unity.UIWidgets.rendering;
+using Unity.UIWidgets.ui;
+using Unity.UIWidgets.widgets;
+
+namespace Unity.UIWidgets.plugins.raycast {
+    class RaycastableBox : SingleChildRenderObjectWidget {
+        public RaycastableBox(
+            Key key = null,
+            Widget child = null
+        ) : base(key, child) {
+            this.windowHashCode = Window.instance.GetHashCode();
+        }
+
+        readonly int windowHashCode;
+
+        public override RenderObject createRenderObject(BuildContext context) {
+            return new RenderRaycastableBox(
+                windowHashCode: this.windowHashCode,
+                widget: this
+            );
+        }
+
+        public override Element createElement() {
+            return new _RaycastableBoxRenderElement(this.windowHashCode, this);
+        }
+    }
+
+    class RenderRaycastableBox : RenderProxyBox {
+        public RenderRaycastableBox(
+            int windowHashCode,
+            RenderBox child = null,
+            RaycastableBox widget = null
+        ) : base(child) {
+            this.widget = widget;
+            this.windowHashCode = windowHashCode;
+        }
+
+        readonly int windowHashCode;
+        RaycastableBox widget;
+
+        public override void detach() {
+            base.detach();
+            this.markNeedsPaint();
+        }
+
+
+        public override void paint(PaintingContext context, Offset offset) {
+            // Debug.Log($"[RenderRaycastableBox] Paint {this.widget.GetHashCode()}: {this.size}@{offset}");
+            RaycastManager.UpdateSizeOffset(this.widget.GetHashCode(), (int) this.windowHashCode, this.size, offset);
```

Comment:
should we cache the widget's hashcode somewhere (or making sure that this hashcode has been cached in this.widget) ? calculate the hashcode every frame seems to be costly ?